File: go/internal/api/provider_test.go
package api

import (
	"reflect"
	"testing"
)

// 三层合并顺序：wellKnown → provider 覆盖 → model 覆盖（后者胜）。
func TestResolveCapabilitiesLayerOrder(t *testing.T) {
	// deepseek 基线：EffortFormat = reasoning_effort
	base := ResolveCapabilities("deepseek", nil, nil)
	if base.EffortFormat != EffortReasoningEffort {
		t.Fatalf("deepseek 基线 EffortFormat = %q, want %q", base.EffortFormat, EffortReasoningEffort)
	}

	// provider 层覆盖
	providerOv := &CapabilitiesOverride{EffortFormat: ptr(EffortNone)}
	got := ResolveCapabilities("deepseek", providerOv, nil)
	if got.EffortFormat != EffortNone {
		t.Errorf("provider 层未生效：%q", got.EffortFormat)
	}

	// model 层覆盖 provider 层（后者胜）
	modelOv := &CapabilitiesOverride{EffortFormat: ptr(EffortOutputConfig)}
	got = ResolveCapabilities("deepseek", providerOv, modelOv)
	if got.EffortFormat != EffortOutputConfig {
		t.Errorf("model 层未覆盖 provider 层：%q", got.EffortFormat)
	}
}

// 未声明字段必须穿透到前一层（nil ≠ 零值）。
func TestUndeclaredFieldsFallThrough(t *testing.T) {
	// 只覆盖 EffortFormat，其余应保持 deepseek 基线
	got := ResolveCapabilities("deepseek", &CapabilitiesOverride{EffortFormat: ptr(EffortNone)}, nil)

	if got.PrefixCacheStrategy != CacheDeepSeekNative {
		t.Errorf("未声明字段被清零：PrefixCacheStrategy = %q", got.PrefixCacheStrategy)
	}
	if !got.PreservedThinkingProtocol {
		t.Error("未声明字段被清零：PreservedThinkingProtocol")
	}
	if !got.HasToolJSONInContentBug {
		t.Error("未声明字段被清零：HasToolJSONInContentBug")
	}
	if got.MapUsage == nil {
		t.Error("未声明字段被清零：MapUsage")
	}
}

// 反证：StripParams 空切片必须与「未声明」同义，不得清空前层。
//
// 若把 applyOverrides 的 `len(ov.StripParams) > 0` 改成 `ov.StripParams != nil`，
// 本测试变红——且会让内置预设的剥离清单静默失效（top_k/cache_control 重新进请求）。
func TestStripParamsEmptyIsNoOpinion(t *testing.T) {
	// deepseek 基线含 4 项剥离参数
	base := ResolveCapabilities("deepseek", nil, nil)
	if len(base.StripParams) != 4 {
		t.Fatalf("deepseek 基线 StripParams 应有 4 项，实为 %d: %v", len(base.StripParams), base.StripParams)
	}

	// 空切片覆盖 → 应视为「无意见」，保持基线 4 项
	got := ResolveCapabilities("deepseek", &CapabilitiesOverride{StripParams: []string{}}, nil)
	if len(got.StripParams) != 4 {
		t.Errorf("空切片被当成显式清空——剥离清单静默失效\n  got:  %v\n  want: %v",
			got.StripParams, base.StripParams)
	}

	// 非空切片 → 正常覆盖
	got = ResolveCapabilities("deepseek", &CapabilitiesOverride{StripParams: []string{"only_this"}}, nil)
	if !reflect.DeepEqual(got.StripParams, []string{"only_this"}) {
		t.Errorf("非空切片未覆盖：%v", got.StripParams)
	}
}

// SupportsThinking 的派生规则。
func TestSupportsThinkingDerivation(t *testing.T) {
	cases := []struct {
		name string
		ov   *CapabilitiesOverride
		want bool
	}{
		{
			"thinkingBlock=enabled 派生 true",
			&CapabilitiesOverride{ThinkingBlock: ptr(ThinkingEnabled)},
			true,
		},
		{
			"thinkingBlock=adaptive 派生 true",
			&CapabilitiesOverride{ThinkingBlock: ptr(ThinkingAdaptive)},
			true,
		},
		{
			"effortFormat=reasoning_effort 派生 true",
			&CapabilitiesOverride{EffortFormat: ptr(EffortReasoningEffort)},
			true,
		},
		{
			"effortFormat=output_config 派生 true",
			&CapabilitiesOverride{EffortFormat: ptr(EffortOutputConfig)},
			true,
		},
		{
			"block=none 且 effort=none 派生 false",
			&CapabilitiesOverride{ThinkingBlock: ptr(ThinkingNone), EffortFormat: ptr(EffortNone)},
			false,
		},
		{
			// 只有 block=none，无 effort 信号 → 保留基线（longcat 基线 false）
			"仅 block=none 时保留基线",
			&CapabilitiesOverride{ThinkingBlock: ptr(ThinkingNone)},
			false,
		},
		{
			// 只有 effort=reasoning_effort → 派生 true（覆盖 longcat 的 false 基线）
			"仅 effort 声明可翻转基线",
			&CapabilitiesOverride{EffortFormat: ptr(EffortReasoningEffort)},
			true,
		},
	}

	for _, tc := range cases {
		t.Run(tc.name, func(t *testing.T) {
			got := ResolveCapabilities("longcat", tc.ov, nil)
			if got.SupportsThinking != tc.want {
				t.Errorf("SupportsThinking = %v, want %v", got.SupportsThinking, tc.want)
			}
		})
	}
}

// 未知提供商回退默认能力。
func TestUnknownProviderFallsBackToDefault(t *testing.T) {
	got := ResolveCapabilities("no-such-provider-xyz", nil, nil)
	if got.SupportsThinking {
		t.Error("未知提供商不应声明支持思考")
	}
	if got.PrefixCacheStrategy != CacheNone {
		t.Errorf("未知提供商 PrefixCacheStrategy = %q, want none", got.PrefixCacheStrategy)
	}
	if !got.SupportsCacheControl {
		t.Error("未知提供商默认支持 cache_control（DEFAULT_CAPABILITIES 语义）")
	}
}

// 反证：ResolveCapabilities 不得让调用方通过返回的切片污染全局表。
//
// 若去掉 cloneStrings / EffortCap 深拷，本测试变红——且会造成跨会话污染：
// 一个会话改了 StripParams，另一个会话跟着变。
func TestResolveDoesNotMutateGlobalTable(t *testing.T) {
	before := ResolveCapabilities("deepseek", nil, nil)

	// 篡改第一次调用的返回值
	mutated := ResolveCapabilities("deepseek", nil, nil)
	mutated.StripParams[0] = "TAMPERED"

	after := ResolveCapabilities("deepseek", nil, nil)
	if after.StripParams[0] == "TAMPERED" {
		t.Fatalf("全局表被返回值污染：%v", after.StripParams)
	}
	if !reflect.DeepEqual(before.StripParams, after.StripParams) {
		t.Errorf("两次调用结果不一致：\n  before: %v\n  after:  %v", before.StripParams, after.StripParams)
	}

	// EffortCap 同理（codex 有 EffortCap）
	m1 := ResolveCapabilities("codex", nil, nil)
	m1.EffortCap["max"] = "TAMPERED"
	m2 := ResolveCapabilities("codex", nil, nil)
	if m2.EffortCap["max"] == "TAMPERED" {
		t.Fatalf("EffortCap 全局表被污染：%v", m2.EffortCap)
	}
}

// MapDeepSeekUsage 的字段优先级与点号路径。
func TestMapDeepSeekUsage(t *testing.T) {
	cases := []struct {
		name string
		raw  map[string]any
		want Usage
	}{
		{
			"DeepSeek 原生格式",
			map[string]any{
				"prompt_tokens":            100,
				"completion_tokens":        20,
				"prompt_cache_hit_tokens":  80,
				"prompt_cache_miss_tokens": 20,
			},
			Usage{InputTokens: 100, OutputTokens: 20, CacheReadInputTokens: 80, CacheCreationInputTokens: 20},
		},
		{
			"Anthropic 兼容格式",
			map[string]any{
				"input_tokens":                100,
				"output_tokens":               20,
				"cache_read_input_tokens":     80,
				"cache_creation_input_tokens": 20,
			},
			Usage{InputTokens: 100, OutputTokens: 20, CacheReadInputTokens: 80, CacheCreationInputTokens: 20},
		},
		{
			"嵌套 cached_tokens 路径",
			map[string]any{
				"prompt_tokens":     100,
				"completion_tokens": 20,
				"prompt_tokens_details": map[string]any{
					"cached_tokens": 55,
				},
			},
			Usage{InputTokens: 100, OutputTokens: 20, CacheReadInputTokens: 55},
		},
		{
			"原生字段优先于兼容字段",
			map[string]any{
				"prompt_tokens":           111,
				"input_tokens":            999,
				"completion_tokens":       22,
				"output_tokens":           888,
				"prompt_cache_hit_tokens": 70,
				"cache_read_input_tokens": 777,
			},
			Usage{InputTokens: 111, OutputTokens: 22, CacheReadInputTokens: 70},
		},
		{
			"空 map 得零值",
			map[string]any{},
			Usage{},
		},
		{
			"浮点数值可转换",
			map[string]any{"prompt_tokens": 100.0, "completion_tokens": 20.0},
			Usage{InputTokens: 100, OutputTokens: 20},
		},
	}

	for _, tc := range cases {
		t.Run(tc.name, func(t *testing.T) {
			got := MapDeepSeekUsage(tc.raw)
			if got != tc.want {
				t.Errorf("MapDeepSeekUsage 不符\n  got:  %+v\n  want: %+v", got, tc.want)
			}
		})
	}
}

// 关键提供商的基线能力（防止表被误改）。
func TestWellKnownBaselines(t *testing.T) {
	cases := []struct {
		name            string
		provider        string
		wantCacheStrat  PrefixCacheStrategy
		wantToolJSONBug bool
	}{
		{"deepseek 用原生前缀缓存且有 toolJsonBug", "deepseek", CacheDeepSeekNative, true},
		{"glm 共享 deepseek-native 缓存", "glm", CacheDeepSeekNative, false},
		{"longcat 隐式前缀缓存", "longcat", CacheDeepSeekNative, false},
		{"claude 无前缀缓存", "claude", CacheNone, false},
		{"minimax 无前缀缓存", "minimax", CacheNone, false},
	}

	for _, tc := range cases {
		t.Run(tc.name, func(t *testing.T) {
			got := ResolveCapabilities(tc.provider, nil, nil)
			if got.PrefixCacheStrategy != tc.wantCacheStrat {
				t.Errorf("PrefixCacheStrategy = %q, want %q", got.PrefixCacheStrategy, tc.wantCacheStrat)
			}
			if got.HasToolJSONInContentBug != tc.wantToolJSONBug {
				t.Errorf("HasToolJSONInContentBug = %v, want %v", got.HasToolJSONInContentBug, tc.wantToolJSONBug)
			}
		})
	}
}

// PreservedThinkingProtocol 只由线格式派生自 DeepSeek 的提供商声明——
// 共享 deepseek-native 缓存策略不等于共享该协议（GLM/longcat 有独立推理）。
func TestPreservedThinkingProtocolOnlyDeepSeekDerived(t *testing.T) {
	shouldHave := []string{"deepseek", "mimo"}
	shouldNotHave := []string{"glm", "longcat", "siliconflow", "kimi", "claude"}

	for _, p := range shouldHave {
		if !ResolveCapabilities(p, nil, nil).PreservedThinkingProtocol {
			t.Errorf("%s 应声明 PreservedThinkingProtocol", p)
		}
	}
	for _, p := range shouldNotHave {
		if ResolveCapabilities(p, nil, nil).PreservedThinkingProtocol {
			t.Errorf("%s 不应声明 PreservedThinkingProtocol（仅共享缓存策略，推理实现独立）", p)
		}
	}
}

func ptr[T any](v T) *T { return &v }

File: src/api/__tests__/provider.test.ts
import { describe, it } from 'node:test'
import assert from 'node:assert/strict'
import { mapDeepSeekUsage, resolveCapabilities, resolveEffortSupported } from '../provider.js'

describe('DeepSeek provider usage mapping', () => {
  it('maps native DeepSeek cache counters into standard usage fields', () => {
    assert.deepEqual(mapDeepSeekUsage({
      prompt_tokens: 1000,
      completion_tokens: 200,
      prompt_cache_hit_tokens: 875,
      prompt_cache_miss_tokens: 125,
    }), {
      input_tokens: 1000,
      output_tokens: 200,
      cache_read_input_tokens: 875,
      cache_creation_input_tokens: 125,
    })
  })

  it('exposes DeepSeek usage mapping through resolved capabilities', () => {
    const capabilities = resolveCapabilities('deepseek')

    assert.deepEqual(capabilities.mapUsage?.({
      prompt_tokens: 400,
      completion_tokens: 50,
      prompt_cache_hit_tokens: 300,
      prompt_cache_miss_tokens: 100,
    }), {
      input_tokens: 400,
      output_tokens: 50,
      cache_read_input_tokens: 300,
      cache_creation_input_tokens: 100,
    })
  })
})

describe('GLM provider implicit prefix cache', () => {
  it('resolves GLM as a deepseek-native (implicit exact-prefix) cache provider', () => {
    const capabilities = resolveCapabilities('glm')
    assert.equal(capabilities.prefixCacheStrategy, 'deepseek-native')
    assert.ok(capabilities.mapUsage, 'GLM must expose a usage mapping to read cached_tokens')
  })

  it('maps GLM OpenAI-style prompt_tokens_details.cached_tokens into cache_read_input_tokens', () => {
    const capabilities = resolveCapabilities('glm')
    assert.deepEqual(capabilities.mapUsage?.({
      prompt_tokens: 1200,
      completion_tokens: 300,
      prompt_tokens_details: { cached_tokens: 800 },
    }), {
      input_tokens: 1200,
      output_tokens: 300,
      cache_read_input_tokens: 800,
      cache_creation_input_tokens: 0,
    })
  })
})

describe('LongCat provider explicit defaults (W4 — no implicit DEFAULT fallback)', () => {
  it('resolves longcat with response_format disabled — worker repair must run as plain-text re-ask', () => {
    const capabilities = resolveCapabilities('longcat')
    assert.equal(capabilities.supportsResponseFormat, false, 'LongCat API has no response_format — json-mode repair unusable')
    assert.equal(capabilities.supportsCacheControl, false, 'no cache_control breakpoints — implicit server-side prefix caching')
    assert.ok(capabilities.stripParams.includes('cache_control'))
    assert.equal(capabilities.prefixCacheStrategy, 'deepseek-native')
    assert.ok(capabilities.mapUsage, 'must read cached_tokens for free cache hits')
  })
})

describe('Aggregator / relay providers — WELL_KNOWN_DEFAULTS', () => {
  it('siliconflow: thinkingBlock enabled + deepseek-native prefix cache', () => {
    const caps = resolveCapabilities('siliconflow')
    assert.equal(caps.supportsThinking, true)
    assert.equal(caps.thinkingBlockType, 'enabled')
    assert.equal(caps.effortFormat, 'reasoning_effort')
    assert.equal(caps.prefixCacheStrategy, 'deepseek-native', 'server-side implicit exact-prefix cache')
    assert.ok(caps.stripParams.includes('cache_control'))
    assert.ok(caps.mapUsage, 'reads cached_tokens via DeepSeek-shaped usage')
  })

  it('dashscope: thinkingBlock enabled + response_format supported (OpenAI-compatible endpoint)', () => {
    const caps = resolveCapabilities('dashscope')
    assert.equal(caps.supportsThinking, true)
    assert.equal(caps.thinkingBlockType, 'enabled')
    assert.equal(caps.supportsResponseFormat, true)
    assert.equal(caps.prefixCacheStrategy, 'none', 'OpenAI-compat endpoint does not accept cache_control')
  })

  it('openrouter: no thinking block — only reasoning_effort passthrough', () => {
    const caps = resolveCapabilities('openrouter')
    assert.equal(caps.thinkingBlockType, 'none', 'thinking block passthrough is unstable across the fleet')
    assert.equal(caps.effortFormat, 'reasoning_effort')
    assert.equal(caps.supportsThinking, true)
  })

  it('relay: generic one-api/new-api template — reasoning_effort passthrough, no thinking block', () => {
    const caps = resolveCapabilities('relay')
    assert.equal(caps.thinkingBlockType, 'none')
    assert.equal(caps.effortFormat, 'reasoning_effort')
    assert.equal(caps.prefixCacheStrategy, 'none')
  })
})

describe('resolveCapabilities — override precedence and thinking derivation', () => {
  it('provider overrides win over WELL_KNOWN defaults', () => {
    const caps = resolveCapabilities('deepseek', { cacheControl: true })
    assert.equal(caps.supportsCacheControl, true, 'explicit cacheControl:true overrides deepseek WELL_KNOWN=false')
    // Unmentioned fields keep WELL_KNOWN values
    assert.equal(caps.thinkingBlockType, 'enabled')
    assert.equal(caps.prefixCacheStrategy, 'deepseek-native')
  })

  it('model overrides win over provider overrides', () => {
    const caps = resolveCapabilities(
      'dashscope',
      { thinkingBlock: 'enabled' },
      { thinkingBlock: 'none', effortFormat: 'none' },
    )
    assert.equal(caps.thinkingBlockType, 'none', 'model-level override takes precedence')
    assert.equal(caps.effortFormat, 'none')
  })

  it('user declaring thinkingBlock=enabled flips supportsThinking=true even for no-thinking providers', () => {
    const caps = resolveCapabilities('longcat', { thinkingBlock: 'enabled' })
    assert.equal(caps.supportsThinking, true)
    assert.equal(caps.thinkingBlockType, 'enabled')
  })

  it('user declaring both block=none and effort=none flips supportsThinking=false', () => {
    const caps = resolveCapabilities('deepseek', {
      thinkingBlock: 'none',
      effortFormat: 'none',
    })
    assert.equal(caps.supportsThinking, false)
  })

  it('user declaring effortFormat alone flips supportsThinking=true and leaves other fields intact', () => {
    const caps = resolveCapabilities('relay', { effortFormat: 'reasoning_effort' })
    assert.equal(caps.supportsThinking, true)
    assert.equal(caps.thinkingBlockType, 'none', 'user did not override block — keeps WELL_KNOWN')
  })

  it('empty stripParams ([]) is treated as "not declared" — WELL_KNOWN list applies (阻断 4)', () => {
    const caps = resolveCapabilities('deepseek', { stripParams: [] })
    assert.ok(caps.stripParams.includes('cache_control'), '空数组回退 WELL_KNOWN 剥离清单，恢复旧模型语义')
  })

  it('effortCap override replaces WELL_KNOWN cap wholesale', () => {
    const caps = resolveCapabilities('kimi', { effortCap: { max: 'medium' } })
    assert.deepEqual(caps.effortCap, { max: 'medium' })
  })

  // 官方 Kimi Code 文档（models.html）：k3 / k3-256k 的 reasoning_effort 支持
  // low|high|max（默认 high），第三方工具传 max 会映射到 max。此前的
  // effortCap {max:'high'} 会把用户显式选的 max 静默降档——K3 旗舰的 max 档
  // 因此永远发不出去。K2.7 Code（kimi-for-coding）是 Thinking:ON，无档位。
  it('kimi 不再钳制 max——K3 系原生支持 low/high/max', () => {
    const caps = resolveCapabilities('kimi')
    assert.equal(caps.effortCap?.max, undefined, 'kimi 不得再把 max 降成 high（K3/K3-256K 原生支持 max）')
  })

  it('resolving with overrides never mutates the shared WELL_KNOWN table', () => {
    resolveCapabilities('deepseek', { cacheControl: true, thinkingBlock: 'none' })
    const fresh = resolveCapabilities('deepseek')
    assert.equal(fresh.supportsCacheControl, false, 'WELL_KNOWN deepseek cacheControl must stay false')
    assert.equal(fresh.thinkingBlockType, 'enabled', 'WELL_KNOWN deepseek thinkingBlock must stay enabled')
  })
})

describe('preservedThinkingProtocol — DeepSeek wire-protocol family', () => {
  it('declared only for DeepSeek-derived providers, not prefix-cache siblings', () => {
    assert.equal(resolveCapabilities('deepseek').preservedThinkingProtocol, true)
    assert.equal(resolveCapabilities('mimo').preservedThinkingProtocol, true)
    // Share deepseek-native prefix cache but have independent reasoning —
    // must NOT inherit the preserved-thinking wire protocol.
    assert.notEqual(resolveCapabilities('glm').preservedThinkingProtocol, true)
    assert.notEqual(resolveCapabilities('longcat').preservedThinkingProtocol, true)
    assert.notEqual(resolveCapabilities('siliconflow').preservedThinkingProtocol, true)
    assert.notEqual(resolveCapabilities('mimo-api').preservedThinkingProtocol, true)
  })

  it('overridable via capabilities (pro spark preset / user config path)', () => {
    const caps = resolveCapabilities('unknown-relay', { preservedThinkingProtocol: true })
    assert.equal(caps.preservedThinkingProtocol, true)
    const off = resolveCapabilities('deepseek', { preservedThinkingProtocol: false })
    assert.equal(off.preservedThinkingProtocol, false)
  })
})

describe('resolveEffortSupported — 会话内调档是否真能上线', () => {
  it('未知 provider 默认无档位通道（effortFormat none）→ false', () => {
    assert.equal(resolveEffortSupported('my-relay', { protocol: 'openai', thinking: 'enabled' }), false)
  })

  it('provider 级 / 模型级 capability 声明都能把通道打开', () => {
    assert.equal(resolveEffortSupported(
      'my-relay', { protocol: 'openai', thinking: 'enabled', capabilities: { effortFormat: 'reasoning_effort' } },
    ), true)
    assert.equal(resolveEffortSupported(
      'my-relay', { protocol: 'openai', thinking: 'enabled' }, { effortFormat: 'reasoning_effort' },
    ), true)
  })

  it('已知预设（deepseek）继承 reasoning_effort 通道', () => {
    assert.equal(resolveEffortSupported('deepseek', { protocol: 'openai', thinking: 'enabled' }), true)
  })

  it('openai-responses 协议直接写 reasoning.effort，自定义 provider 也支持', () => {
    assert.equal(resolveEffortSupported('my-relay', { protocol: 'openai-responses', thinking: 'enabled' }), true)
  })

  it('anthropic 协议与 thinking disabled 都算不支持（前者预算固定，后者不发该字段）', () => {
    assert.equal(resolveEffortSupported('my-claude', { protocol: 'anthropic', thinking: 'enabled' }), false)
    assert.equal(resolveEffortSupported('deepseek', { protocol: 'openai', thinking: 'disabled' }), false)
  })
})

describe('Grok (xAI) — reasoning_effort 透传', () => {
  it('能力位：thinkingBlock none + effortFormat reasoning_effort + off/max 映射', () => {
    const caps = resolveCapabilities('grok')
    assert.equal(caps.supportsThinking, true)
    assert.equal(caps.thinkingBlockType, 'none')
    assert.equal(caps.effortFormat, 'reasoning_effort')
    assert.deepEqual(caps.effortCap, { off: 'low', max: 'xhigh' })
    assert.ok(caps.stripParams.includes('frequency_penalty'), 'xAI 推理模型拒收 frequency_penalty')
    assert.ok(caps.stripParams.includes('presence_penalty'), 'xAI 推理模型拒收 presence_penalty')
    assert.ok(caps.stripParams.includes('stop'), 'xAI 推理模型拒收 stop')
    assert.equal(resolveEffortSupported('grok', { protocol: 'openai', thinking: 'enabled' }), true)
  })

  it('provider 级 capabilities 覆盖不破坏档位映射', () => {
    const caps = resolveCapabilities('grok', { cacheControl: false, stripParams: [] })
    assert.equal(caps.effortFormat, 'reasoning_effort')
    assert.deepEqual(caps.effortCap, { off: 'low', max: 'xhigh' })
  })
})

describe('StepFun provider capabilities', () => {
  it('stepfun：reasoning_effort 三档 + 官方上限 high（max 降级）+ 服务端隐式提示缓存', () => {
    const caps = resolveCapabilities('stepfun')
    assert.equal(caps.supportsThinking, true)
    assert.equal(caps.thinkingBlockType, 'none', '官方只走 reasoning_effort，不发 thinking block')
    assert.equal(caps.effortFormat, 'reasoning_effort')
    assert.deepEqual(caps.effortCap, { max: 'high', off: 'low' })
    assert.equal(caps.prefixCacheStrategy, 'deepseek-native', '提示缓存是服务端隐式的（无客户端断点）')
    assert.equal(caps.supportsCacheControl, false)
    assert.equal(caps.supportsResponseFormat, true, '官方支持 JSON Mode 与 JSON Schema')
    assert.equal(
      caps.preservedThinkingProtocol,
      undefined,
      '不是 DeepSeek 系线协议——不得套用 reasoning_content 回显与中文思考后缀',
    )
    assert.ok(caps.stripParams.includes('cache_control'), 'OpenAI 兼容端点不吃 Anthropic 的缓存断点')
    assert.equal(resolveEffortSupported('stepfun', { protocol: 'openai', thinking: 'enabled' }), true)
  })
})
